Wrappers/Python/cil/optimisation/functions/L1Norm.py

```diff
@@ -37,10 +36,10 @@ def soft_shrinkage(x, tau, out=None):
     --------
     the value of the soft-shrinkage operator at x: DataContainer.
     """
-    if  id(x)==id(out):
-        raise InPlaceError(message="The soft_shrinkage function cannot be used in place" )
-
     should_return = False
+    # get the sign of the input  
+    dsign = np.exp(1j*np.angle(x.as_array())) if isinstance(x.dtype, np.complexfloating) else x.sign() 
```

Comment:
```suggestion
    dsign = np.exp(1j*np.angle(x.as_array())) if issubclass(x.dtype, np.complexfloating) else x.sign()
```

Reply:
It was failing unit tests with `issubclass`:

![image](https://github.com/TomographicImaging/CIL/assets/43645617/16d67304-662f-4402-9250-66a4babf543b)

Reply:
Thanks both